packages/dds/tree/docs/main/merge-semantics.md

```diff
@@ -0,0 +1,519 @@
+(the node being at the chosen destination)
+so long as the preconditions of the move were met.
+
+Note that such conditional postconditions would be different from additional preconditions.
+For example, adding a precondition that the moved node must not have been concurrently removed would,
+in a situation where that did occur,
+render such a move _and the whole transactions it figures in_ invalid.
+By contrast, making the postcondition conditional on the node not having been concurrently removed would simply render the move ineffective
+but not adversely affect the rest of the transaction.
+
+As with minimal preconditions, we may support more variations of our current set of edits in the future.
+
+This prioritization is partially guided by the same motivation that guided our choice with minimal preconditions:
+it makes for a greater set of possible edits when combined with [constraints](#constraints).
+
+The other motivation is that simpler postconditions make authoring transactions much more approachable.
+This is because they reduce the number of possible states the document could be in after each edit that makes up the transaction.
+
+For example, consider an application that allows the end user to select a set of sticky notes across several pages,
+and group all of the selected notes under a new page, assigning to each one an ordinal number based on the selection order.
+This functionality effectively allows a user to make a numbered list out of a set of N sticky nodes.
+
+This can be achieved by writing a transaction that loops through the N selected sticky notes in the order they were selected,
+assigning ordinals incrementally and moving each one to the end of the new page.
+
+By the time this transaction is sequenced and applied,
+concurrent edits may have affected the relevant sticky notes in different ways:
+some of them may have been assigned different ordinals,
+some of them may have been moved/removed,
+and some of their parent pages may have been moved/removed.
+Despite that, `SharedTree`'s current merge semantics guarantee that by the end of the transaction,
+all of the relevant sticky notes will reside in the new page and that their ordinals will be assigned in order from 1 to N.
+If any of the concurrent changes had the power to prevent the relevant notes from being moved by the transaction or from being annotated with the ordinals,
+then the transaction may lead to a state where the new page only contains a subset of the selected notes, and their ordinals may have gaps, may not be unique, or may be out of order.
+In other words, it would be very difficult to make any kind of valuable claim about the effect of the transaction,
+and end users would experience some very confusing outcomes.
+
+To get a more general sense of the positive effect of simpler postconditions,
+we can model transactions as being composed of `N` edits (`e1` through `eN`),
+where the effect of each edit `ei` is one out of a set whose size is `ki`.
```

Comment:
```suggestion
where the effect of each edit `ei` is one out of a set whose size is `ki` where `ki` is the number of postconditions for `ei`.
```

Reply:
Postconditions are always all true, so having multiple postcondions is not the problem. The probem is having a single (or many, but that's not material here) postcondition that is itself disjunctive.
This making me realize I need to make this clearer at the start of the section. My use of the plural is confusing.